### ISSUE TITLE
[NixOS] Xfce and i3 combination broken 

### ISSUE THREAD

**Issue by brodul** [NixOS/nixpkgs#10634]
I am using  15.09 channel. 

### Problem

When the desktop initializes the `xfce4-session` and the `xfce4-power-manager` are not started. 
Meaning that the events like 'Suspend on lid close' are not working (no power manager). And the `nm-applet` is not started ( no xfce startup programs)

### Steps to reproduce:

- add this lines to `configuration.nix`

```
  services.xserver.desktopManager.xfce.enable = true;
  services.xserver.windowManager.i3.enable = true;
  services.xserver.windowManager.default = "i3";
```
- rebuild switch the configuration
- restart X server

### Possible solutions/workarounds:
- deleting xfce4 user settings (goodbye state) (does not help)
- configuring .i3 to start `nm-applet` (works)
- configuring .i3 to start `xfce4-power-manager` (does not work (lack of permissions in order to connect to DBUS) )

**Comment by domenkozar** [comment]
Do you by any chance have `~/.xsession` file? That will override xfce startup.

**Comment by brodul** [comment]
I have no `~/.xsession` nor `~/.xinitrc`.

**Comment by domenkozar** [comment]
Hmm, OK. Is this something that started recently? cc @vcunat

**Comment by vcunat** [comment]
I don't know, as I just use `desktopManager.xfce.enable = true;` and start WM from my xfce session (XMonad). I'm not even certain `windowManager` is a good way to configure such things, as it seems very user-specific to me.

**Comment by vcunat** [comment]
suspend-on-lid-close: does it really not work? IIRC the current config should be that it does suspend by default unless a desktop/power manager overrides it.

**Comment by vcunat** [comment]
Side note: multi-user systems aren't very common nowadays, but still, it feels wrong to me to set such things in `configuration.nix`. Maybe we should allow them (also) in `~/.nixpkgs/config.nix` or a similar location (in future), even custom `desktopManager`, etc.

**Comment by brodul** [comment]
@vcunat Suspend on lid works for the first time. Then stops working.
I know it's a bit optimistic to expect xfce and i3 work out of the box (just by configuring NixOS), but it did in 14.12 release and before.

Anyway I fixed it with the help @domenkozar . I was comparing the setup with @domenkozar which is using the unstable all the time.
So now I reproduced his settings and now it works for me.
My `configuration.nix` is the same.

I just added `.xsession` (don't forget to `chmod +x` it) to my home which overrides xserver initialization.  
```
startxfce4&
dunst&
nm-applet&
exec i3
```
It works for me. But the behavior definitely changed from 14.12 to 15.09 .
